### PR TITLE
Generalize Graph

### PR DESCRIPTION
**Pull request by dustin**
We have two types of graphs and a conversion between them.  Now we can have N types and conversions among all of them.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -14,9 +14,12 @@ import           Data.Text            (Text)
...
 import           Data.Tree.View       (drawTree)
-import           Options.Applicative  (Parser, argument, command, customExecParser, fullDesc, help, helper, hsubparser,
-                                       info, long, metavar, prefs, progDesc, short, showDefault, showHelpOnError, some,
-                                       str, strOption, switch, value, (<**>))
+import           Options.Applicative  (Parser, argument, command,
+                                       customExecParser, fullDesc, help, helper,
+                                       hsubparser, info, long, metavar, prefs,
+                                       progDesc, short, showDefault,
+                                       showHelpOnError, some, str, strOption,
+                                       switch, value, (<**>))
```
Comment (dustin) [inline app/Main.hs:22]: Annoying unintentional formatter change

File: src/Graphex.hs [hunk 2]
```diff
@@ -20,61 +20,61 @@ import           Graphex.LookingGlass
...
 --
 -- This is a short path, but the important part is that it represents how connectivy works.
-why :: Graph -> Text -> Text -> [Text]
+why :: Ord a => Graph a -> a -> a -> [a]
 why m from to = fromMaybe [] $ findFirst bfsOn head (\ks -> (:ks) <$> (Set.toList . directDepsOn m . head) ks) [from] ((== to) . head)
 
 -- | Find all paths between two modules as a restricted graph of the intersection of
 -- reachable nodes from the start and to the end.
-allPathsTo :: Graph -> Text -> Text -> Graph
+allPathsTo :: (NFData a, Ord a) => Graph a -> a -> a -> Graph a
```
Comment (dustin) [inline src/Graphex.hs:58]: The NFData constraints are a bit of a leak.

Comment (ramirez7) [reply src/Graphex.hs:58]: Yeah that's a little unfortunate, but not a big deal.

**Review by ramirez7 — APPROVED** [review]
:metal:

**Pull request merged**